### ISSUE TITLE
old URLs have broken

### ISSUE THREAD

**Issue by ljharb** [mdn/yari#3012]
https://developer.mozilla.org/en-US/DOM/element.addEventListener 404s but should probably 301 to https://developer.mozilla.org/en-us/docs/Web/API/EventTarget/addEventListener

I’m not sure when it broke; the old URL has been in an IRC factoid for the better part of a decade.

**Comment by peterbe** [comment]
https://github.com/mdn/yari/pull/2872 does not appear to have worked. 
```
▶ curl -I https://developer.mozilla.org/en-US/DOM/window.getComputedStyle
HTTP/2 404
```
I don't know about that ending (`DOM/window.getComputedStyle`) but why didn't the fundamental redirects at least try to go to `docs/DOMS/window.getComputedStyle` first. 
I.e. I'd expect...

1. A 301 (or 302!) redirect to `https://developer.mozilla.org/en-US/docs/DOM/window.getComputedStyle` first
2. A 200 or a 404 (or even a 301 if that URL was part of the `_redirects.txt`)

And whilst we're on the subject, in https://github.com/mdn/yari/issues/2863#issuecomment-783706551 I did a bunch of research by downloading the "Page not found" pageviews from Google Analytics for the last 4 days and from that I concluded that almost 92% of 404s could be fixed if we just injects `/docs/` between the locale and the slug. **78%** of these come from the 10 most frequently seen prefixes. In particular the list is as follows:

```
1 JavaScript           47 28.3%
2 DOM                  41 24.7%
3 CSS                  16 9.6%
4 HTML                 5  3.0%
5 SVG                  4  2.4%
6 Tools                4  2.4%
7 XMLHttpRequest       4  2.4%
8 Web                  3  1.8%
9 Using_files_from_web_applications 2  1.2%
10 AJAX                 2  1.2%
11 DragDrop             2  1.2%
```

So perhaps we can write a custom regex just for these popular ones and even though it hurts to hardcode things, it'll sweep away 78% of 92% of all Page not founds.

**Comment by escattone** [comment]
@peterbe See https://github.com/mdn/yari/pull/3013

**Comment by escattone** [comment]
I merged #3013 and updated our CDN behaviors on `dev`, `stage`, and `prod` to use the latest version of our Lambda@Edge origin-request function. The  URL https://developer.mozilla.org/en-US/DOM/element.addEventListener works now, as should all of the `/en-US/DOM/...` family of URLs (e.g., https://developer.mozilla.org/en-US/DOM/DOM_Reference).
